packages/patrol/lib/src/extensions.dart

```diff
@@ -0,0 +1,16 @@
+import 'package:meta/meta.dart';
+// ignore: implementation_imports
+import 'package:test_api/src/backend/invoker.dart';
+
+/// Provides convenience methods for [Invoker].
+@internal
+extension InvokerX on Invoker {
+  /// Returns the full name of the current test (names of all ancestor groups +
+  /// name of the current test).
+  String fullCurrentTestName() {
+    final parentGroupName = liveTest.groups.last.name;
+    final testName = liveTest.individualName;
+
+    return '$parentGroupName $testName';
```

Comment:
Is there any requirement that this should be unique in a test suite?

Reply:
Yeah, but it's a requirement that's already enforced by `package:test` itself.

Reply:
is it?

```sh
➜ cat test/test_test.dart
import 'package:test/expect.dart';
import 'package:test/scaffolding.dart';

void main() {
  test('group test', () {
    expect(true, true);
  });
  group('group', () {
    test('test', () {
      expect(true, true);
    });
  });
}

➜ dart test test/test_test.dart
00:00 +2: All tests passed!
```

Reply:
you're right. thanks

We could use `_` to separate groups. It'll be required anyway if we want to reconstruct the original test suite hierarchy later.

Reply:
Do you plan to address it now or later?

Reply:
Later, as part of #1502.